File: DayFour/Java/TicTac.java
import java.util.Scanner;

public class TicTac {

    public static void main(String [] args) {

    Scanner scanner = new Scanner(System.in);

    System.out.println("Welcome to X and O game!!!");

    System.out.println("Rules -> You are only allowed to write either X or O");

    System.out.print("Enter your name: ");
    String name = scanner.nextLine();

    for(int counter = 0; counter < 9; counter++) {

      System.out.print("Enter a letter either X or O: ");
      String input = scanner.next();
      char choice = input.charAt(0);
      boolean validInput = choice == 'X' || choice == 'O';

      if (choice != 'X' && choice != 'O') {
          System.out.println("Invalid input. Please enter a letter either X or O.");
        }
      else {
        int column;
        while (true) {
          System.out.print("Which column do you want to place your letter (1-9): ");
          column = scanner.nextInt();

          if (column >= 1 && column <= 9) {
             break;
            }
         else {
           System.out.println("Invalid column. Please enter a number between 1 and 9");
          }
         }
        }


    }

 }

}

File: DayFour/JavaScript/tictac.js
const prompt = require('prompt-sync')();

console.log("Welcome to X and O game!!!");
console.log("Rules -> You are only allowed to write either X or O");

const name = prompt("Enter your name: ");

for (let counter = 0; counter < 9; counter++) {
    const input = prompt("Enter a letter either X or O: ");
    const choice = input.trim().toUpperCase();
    const validInput = choice === 'X' || choice === 'O';

    if (!validInput) {
        console.log("Invalid input. Please enter a letter either X or O.")
        continue;
    }

    let column;
    while (true) {
         const column = prompt("Which column do you want to place your letter (1-9): ");
         column = parseInt(column);
         if (column >= 1 && column <= 9) {
             break;
          }
         else {
           console.log("Invalid column. Please enter a number between 1 and 9");
         }
       }

 }

console.log("Game Over!!!");